### PR TITLE
COG-1226 user dataModel locked attribute

### PR DESCRIPTION
**Pull request by sorych**
### JIRA Issue Link
- [COG-1226: user dataModel locked attribute](https://osi-cwds.atlassian.net/browse/COG-1226)

### Tests
- [ ] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/idm/dto/User.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
 
   private String phoneExtensionNumber;
 
+  private boolean locked;
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/dto/User.java]: Why enabled is Boolean and locked boolean? I think there should be consistency in datatypes.

File: src/main/java/gov/ca/cwds/idm/service/cognito/attribute/CustomUserAttribute.java [hunk 2]
```diff
@@ -7,7 +7,8 @@
   COUNTY("custom:County"),
   RACFID_CUSTOM("custom:RACFID"),
   RACFID_CUSTOM_2("custom:RACFId"),
-  PHONE_EXTENSION("custom:PhoneExtension");
+  PHONE_EXTENSION("custom:PhoneExtension"),
+  IS_LOCKED("custom:locked");
```
Comment (dmytro-palczewski-osi) [inline src/main/java/gov/ca/cwds/idm/service/cognito/attribute/CustomUserAttribute.java]: my be better "custom:Locked"?

**Comment by dmytro-palczewski-osi** [comment]
Should not we assign locked=false attribute at user creation?

File: src/main/java/gov/ca/cwds/idm/dto/User.java (hunk 1 above)
Comment (sorych) [reply src/main/java/gov/ca/cwds/idm/dto/User.java]: enabled is an Object in Cognito api and "locked" is an attribute of our user that cannot be null by our design.

File: src/main/java/gov/ca/cwds/idm/service/cognito/attribute/CustomUserAttribute.java (hunk 2 above)
Comment (sorych) [reply src/main/java/gov/ca/cwds/idm/service/cognito/attribute/CustomUserAttribute.java]: the name is exactly the same as we have it added to userpools on all envs

**Comment by sorych** [comment]
@dmytro-palczewski-osi that may add more transparency but generally not required. no value stored in cognito is interpreted as "false" by our mapping.

**Comment by dmytro-palczewski-osi** [comment]
"no value stored in cognito is interpreted as "false" by our mapping."
What about Lambda function? May we be sure that it will treat absence of locked attribute as false?

**Comment by sorych** [comment]
@dmytro-palczewski-osi I will make this notice to developers that currently work on lambda. Anyway, as we are modifying existent userpools no value are present for previously created users so we need to build logic correspondently.

**Pull request merged**